File: Dev/C++/Mascaret/src/VEHA/Entity/RotationVector.cpp
#include "VEHA/Entity/RotationVector.h"
#include "MascaretApplication.h"

namespace VEHA
{
RotationVector::RotationVector(double x,double y,double z,double a)
: ValueSpecification(MascaretApplication::getInstance()->getModel()->getBasicType("rotation")), x(x),y(y),z(z),angle(a)
{
}

RotationVector::RotationVector(const Vector3& v,double a)
: ValueSpecification(MascaretApplication::getInstance()->getModel()->getBasicType("rotation")),x(v.x),y(v.y),z(v.z),angle(a)
{
}
RotationVector::RotationVector(const string& str)
: ValueSpecification(MascaretApplication::getInstance()->getModel()->getBasicType("rotation"))
{
        std::vector<std::string> strs;
        boost::split(strs,str,boost::is_any_of("\t "));
        if(strs.size()>3)
        {
                x=boost::lexical_cast<double>(strs[0]);
                y=boost::lexical_cast<double>(strs[1]);
                z=boost::lexical_cast<double>(strs[2]);
                angle=boost::lexical_cast<double>(strs[3]);
        }
}
shared_ptr<ValueSpecification> RotationVector::clone() const
{
        shared_ptr<ValueSpecification> value(new RotationVector(x,y,z,angle));
        return value;
}
RotationVector& RotationVector::operator=(const RotationVector& value)
{
        this->x=((const RotationVector&)value).x;
        this->y=((const RotationVector&)value).y;
        this->z=((const RotationVector&)value).z;
        this->angle=((const RotationVector&)value).angle;
        return *this;
}
}

File: Dev/CS/Mascaret/Mascaret/VEHA/Entity/RotationVector.cs
using System;
using System.Collections.Generic;


namespace Mascaret
{
    public class RotationVector : ValueSpecification
    {
        public double x, y, z, angle;

        //Default values x=1, y=0, z=0, a=0
        public RotationVector(double x, double y, double z, double a)
            : base(MascaretApplication.Instance.Model.getBasicType("rotation"))
        {
            this.x = x;
            this.y = y;
            this.z = z;
            this.angle = a;
        }

        public RotationVector(Vector3 v, double a)
            : base(MascaretApplication.Instance.Model.getBasicType("rotation"))
        {
            this.x = v.x;
            this.y = v.y;
            this.z = v.z;
            this.angle = a;
        }

        public RotationVector(string str)
            : base(MascaretApplication.Instance.Model.getBasicType("rotation"))
        {
            string[] strs = str.Split(char.Parse("\t "));
            if (strs.Length == 4)
            {
                x = Double.Parse(strs[0]);
                y = Double.Parse(strs[1]);
                z = Double.Parse(strs[2]);
                angle = Double.Parse(strs[3]);
            }
            else
                System.Console.WriteLine("string is not formated correctly");
        }

        public override ValueSpecification clone()
        {
            RotationVector valueSpec = new RotationVector(x, y, z, angle);
            return (ValueSpecification)valueSpec;
        }
    }
}

